code/game/machinery/computer/arcade.dm

```diff
@@ -77,22 +77,28 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 		return INITIALIZE_HINT_QDEL
 	Reset()
 
-/obj/machinery/computer/arcade/proc/prizevend(mob/user)
-	SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "arcade", /datum/mood_event/arcade)
-	if(prob(0.0001)) //1 in a million
-		new /obj/item/gun/energy/pulse/prize(src)
-		visible_message("<span class='notice'>[src] dispenses.. woah, a gun! Way past cool.</span>", "<span class='notice'>You hear a chime and a shot.</span>")
-		user.client.give_award(/datum/award/achievement/misc/pulse, user)
-		return
+/obj/machinery/computer/arcade/proc/prizevend(mob/user, prizes = 1)
+	if(user.mind.get_skill_level(/datum/skill/gaming) >= SKILL_LEVEL_LEGENDARY && HAS_TRAIT(user, TRAIT_GAMERGOD))
+		visible_message("<span class='notice'>[user] inputs an intense cheat code!",\
+		"<span class='notice'>You hear a flurry of buttons being pressed.</span>")
+		say("CODE ACTIVATED: EXTRA PRIZES.")
+		prizes *= 2
+	for(var/i = 0, i < prizes, i++)
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "arcade", /datum/mood_event/arcade)
+		if(prob(0.0001)) //1 in a million
```

Comment:
If the chance is so low, is it even worth to add it? Something feasible, maybe 1% or even 0.5% so this can happen.